### PR TITLE
Add Transfers and IBAN functionality

### PR DESCRIPTION
**Pull request by EAddario**
Add missing API v2 paths and operations
Update dependencies to newer versions
Bug fixes and other minor changes

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,16 +4,16 @@
 
     <groupId>com.currencycloud.currencycloud-java</groupId>
     <artifactId>currencycloud-java</artifactId>
-    <version>0.7.9-SNAPSHOT</version>
+    <version>0.9.1</version>
```
Comment (jeremy-cc) [inline pom.xml]: This should be -SNAPSHOT - the mvn release lifecycle is responsible for promotion itself, not the pom.

File: pom.xml [hunk 2]
```diff
@@ -76,17 +82,21 @@
...
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
-                <version>3.3</version>
+                <version>3.7.0</version>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <configuration>
-                    <source>1.7</source>
-                    <target>1.7</target>
+                    <source>1.8</source>
+                    <target>1.8</target>
```
Comment (jeremy-cc) [inline pom.xml]: This is a breaking change for anyone downstream and feels like it should be part of a major version bump, i.e 0.x.x to 1.0.x

File: src/main/java/com/currencycloud/client/CurrencyCloudClient.java [hunk 3]
```diff
@@ -40,7 +40,7 @@
             "[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}",
             Pattern.CASE_INSENSITIVE
     );
-    private static final String userAgent = "CurrencyCloudSDK/2.0 Java/0.7.8";
+    private static final String userAgent = "CurrencyCloudSDK/2.0 Java/0.9.1";
```
Comment (jeremy-cc) [inline src/main/java/com/currencycloud/client/CurrencyCloudClient.java]: We should define the version elsewhere and then link it to the UserAgent through interpolation.

File: src/test/java/com/currencycloud/client/ErrorTest.java [hunk 4]
```diff
@@ -19,14 +19,14 @@
 @SuppressWarnings({"ThrowableResultOfMethodCallIgnored", "UnnecessaryBoxing"})
 public class ErrorTest extends BetamaxTestSupport {
 
-    private String loginId = "rjnienaber@gmail.com";
-    private String apiKey = "ef0fd50fca1fb14c1fab3a8436b9ecb65f02f129fd87eafa45ded8ae257528f0";
+    private String loginId = "development@currencycloud.com";
+    private String apiKey = "deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef";
```
Comment (jeremy-cc) [inline src/test/java/com/currencycloud/client/ErrorTest.java]: 🤣 👍 though I always preferred 0xcafebabe myself 💃

File: pom.xml (hunk 2 above)
Comment (aomega08) [reply pom.xml]: 👍

File: src/main/java/com/currencycloud/client/CurrencyCloud.java [hunk 5]
```diff
@@ -54,17 +54,22 @@ Account createAccount(
...
-            @Nullable @FormParam("postal_code") String postalCode,
-            @Nullable @FormParam("country") String country,
             @Nullable @FormParam("spread_table") String spreadTable,
             @Nullable @FormParam("identification_type") String identificationType,
-            @Nullable @FormParam("identification_value") String identificationValue
+            @Nullable @FormParam("identification_value") String identificationValue,
+            @Nullable @FormParam("api_trading") Boolean apiTrading,
+            @Nullable @FormParam("online_trading") Boolean onlineTrading,
+            @Nullable @FormParam("phone_trading") Boolean phoneTrading
+
```
Comment (aomega08) [inline src/main/java/com/currencycloud/client/CurrencyCloud.java]: Changing the order of parameters to a core call is a breaking change. Major version dump and documentation rewrite (with special large highlighted box for this change)?

File: README.md [hunk 6]
```diff
@@ -2,45 +2,41 @@
...
 ### 1. Using Maven
 
-To use the Currencycloud SDK in a Maven project, add the following dependency to the project's `pom.xml`:
+To use the Currencycloud SDK in a Maven project, add the following dependency to your project's `pom.xml`:
 
 ```xml
 <dependency>
     <groupId>com.currencycloud.currencycloud-java</groupId>
     <artifactId>currencycloud-java</artifactId>
-    <version>0.7.8</version>
```
Comment (derikvercueil) [inline README.md]: There is an involved process during release where you have to create a snapshot first - due to the way that MAVEN and Nexus work. See https://github.com/CurrencyCloud/currencycloud-java/wiki/Releasing-the-SDK

File: pom.xml (hunk 1 above)
Comment (derikvercueil) [reply pom.xml]: Yep. As per my comment above. See: https://github.com/CurrencyCloud/currencycloud-java/wiki/Releasing-the-SDK

File: src/main/java/com/currencycloud/client/CurrencyCloudClient.java (hunk 3 above)
Comment (EAddario) [reply src/main/java/com/currencycloud/client/CurrencyCloudClient.java]: Agree. Ideally it should be derived from the pom. ToDO

File: src/main/java/com/currencycloud/client/CurrencyCloud.java (hunk 5 above)
Comment (EAddario) [reply src/main/java/com/currencycloud/client/CurrencyCloud.java]: Order changed for URL parameters only. Signature of core call "public Account createAccount(Account account)" in CurrencyCloud.java continues to be same

File: README.md (hunk 6 above)
Comment (EAddario) [reply README.md]: Added **-SNAPSHOT** to suggested version. Release can only be driven by maintainers

File: pom.xml (hunk 1 above)
Comment (EAddario) [reply pom.xml]: Changed to conform with contributing guidelines

File: pom.xml (hunk 2 above)
Comment (EAddario) [reply pom.xml]: Reverted com.github.mmazi.rescu back to v1.8.1 to allow Java 7 support for this release

File: README.md (hunk 6 above)
Comment (jeremy-cc) [reply README.md]: One more thing - is there a specific reason for bumping this to 0.9.1 here?  Would 0.7.9 not make more sense as we are adding API functionality but not changing the manner in which the client library works at all?

**Pull request merged**